lib/engine/game/g_18_esp/game.rb

```diff
@@ -777,11 +777,9 @@ def check_distance(route, visits)
           hexes = route.hexes.reject { |h| self.class::DOUBLE_HEX.include?(h.name) }
           raise GameError, 'Route visits same hex twice' if hexes.size != hexes.uniq.size
 
-          if route.train.id == '2P-0' && !@perm2_ran_aranjuez && route.hexes.none? do |hex|
-               hex.id == ARANJUEZ_HEX
-             end
+          if route.train.id == '2P-0' && !@perm2_ran_aranjuez && !([ARANJUEZ_HEX, MADRID_HEX] - route.hexes.map(&:id)).empty?
```

Comment:
Most of this test is repeated in `check_p2_aranjuez`, so you could wrap it in a method to avoid the repetition.

Reply:
fixed, i still need to check for train 2p-0, otherwise i get this error for every train not running this route. unless there is a better way of doing this